libs/fb-login/src/lib/fb-login.ts

```diff
@@ -16,26 +16,39 @@ export interface UserDetails {
 }
 
 export class FbLogin {
+  private lastIsLoggedIn: boolean;
+  private resolveLoggedIn: () => void;
+
   initState: Promise<boolean>;
+  loggedInPromise: Promise<void>;
   userDetails: UserDetails;
   get isLoggedIn(): boolean {
     return !!this.userDetails;
   }
 
   constructor(private config: FbLoginConfig) {
-    this.initState = fbInitializer(config.fbAppId).then(
-      (authResponse: AuthResponse) => {
-        if (!this.userDetails)
-          this.userDetails = {
-            extraInfo: '',
-            facebookId: '',
-            id: '',
-            name: '',
-          };
-        this.userDetails.token = authResponse ? authResponse.accessToken : null;
-        return true;
-      }
-    );
+    this.loggedInPromise = new Promise(resolveLoggedIn => {
```

Comment:
This promise is resolved when login is done. I need this to pull report list in case that the login is done programmatically right after the app is opened, and not after a button click.